contracts/VaultV3.vy

```diff
@@ -662,52 +638,16 @@ def _max_withdraw(
     return max_assets
 
 @internal
-def _deposit(sender: address, recipient: address, assets: uint256) -> uint256:
+def _deposit(sender: address, recipient: address, assets: uint256, shares: uint256):
     """
-    Used for `deposit` calls to transfer the amount of `asset` to the vault, 
-    issue the corresponding shares to the `recipient` and update all needed 
+    Used for `deposit` and `mint` calls to transfer the amount of `asset` to the vault, 
+    issue the corresponding `shares` to the `recipient` and update all needed 
     vault accounting.
     """
-    assert self.shutdown == False # dev: shutdown
-    
-    amount: uint256 = assets
-    # Deposit all if sent with max uint
-    if amount == max_value(uint256):
-        amount = ERC20(self.asset).balanceOf(msg.sender)
-
-    assert amount <= self._max_deposit(recipient), "exceed deposit limit"
- 
-    # Transfer the tokens to the vault first.
-    self._erc20_safe_transfer_from(self.asset, msg.sender, self, amount)
-    # Record the change in total assets.
-    self.total_idle += amount
-    
-    # Issue the corresponding shares for amount.
-    shares: uint256 = self._issue_shares_for_amount(amount, recipient)
-
-    assert shares > 0, "cannot mint zero"
-
-    log Deposit(sender, recipient, amount, shares)
-
-    if self.auto_allocate:
-        self._update_debt(self.default_queue[0], max_value(uint256), 0)
-
-    return shares
-
-@internal
-def _mint(sender: address, recipient: address, shares: uint256) -> uint256:
-    """
-    Used for `mint` calls to issue the corresponding shares to the `recipient`,
-    transfer the amount of `asset` to the vault, and update all needed vault 
-    accounting.
-    """
-    assert self.shutdown == False # dev: shutdown
-    # Get corresponding amount of assets.
-    assets: uint256 = self._convert_to_assets(shares, Rounding.ROUND_UP)
-
-    assert assets > 0, "cannot deposit zero"
     assert assets <= self._max_deposit(recipient), "exceed deposit limit"
-
+    assert assets > 0, "cannot deposit zero"
+    assert shares > 0, "cannot mint zero"
+ 
     # Transfer the tokens to the vault first.
     self._erc20_safe_transfer_from(self.asset, msg.sender, self, assets)
```

Comment:
Why not use the input parameter `sender` instead of `msg.sender`?
Below in the event Deposit, parameter `sender` is used.

I would either remove the sender parameter and go with `msg.sender` in both places or use the `sender` address for both, to limit confusion.

Reply:
good point

I believe it is actually cheaper to use msg.sender using a memory variable. And clearer so will remove the variable

Reply:
https://github.com/yearn/yearn-vaults-v3/pull/209/commits/5b7591662c845a3d141a715a78f3340d591cb38f